datafusion/physical-plan/src/sorts/cursor.rs

```diff
@@ -284,6 +333,34 @@ impl<T: FieldValues> Cursor for FieldCursor<T> {
         self.offset += 1;
         t
     }
+
+    fn slice(&self, offset: usize, length: usize) -> Self {
+        let FieldCursor {
+            values,
+            offset: _,
```

Comment:
This seems at odds with the behaviour of RowCursor

Reply:
We can remove the data slicing of the underlying `FieldCursor.values`. That slicing is a zero-copy of the underlying [ScalarBuffer](https://github.com/apache/arrow-rs/blob/ab87abdd69ab787fdf247cf36f04abc1fbfa6266/arrow-buffer/src/buffer/scalar.rs#L76) or [GenericByteArray](https://github.com/apache/arrow-rs/blob/ab87abdd69ab787fdf247cf36f04abc1fbfa6266/arrow-array/src/array/byte_array.rs#L325).

Would you prefer a switch to using FieldCursor offsets in the same as the RowCursor?

Reply:
I don't see an issue with slicing the underlying values, my observation is that the following will behave differently between RowCursor and FieldCursor

```
cursor.advance();
cursor.slice(1, 2);
```

In the case of `RowCursor` it will produce a slice that is offset by 2 from the start, whereas FieldCursor will produce one that is only offset by 1? I think...